content/oci/oci.go

```diff
@@ -192,6 +192,22 @@ func (s *Store) Resolve(ctx context.Context, reference string) (ocispec.Descript
 	return desc, nil
 }
 
+func (s *Store) Delete(ctx context.Context, target ocispec.Descriptor) error {
```

Comment:
One tricky thing about deleting is that we need to maintain the predecessors relationship as well. See [Store.Push](https://github.com/oras-project/oras-go/blob/main/content/oci/oci.go#L116-L117) and [Store.Predecessors](https://github.com/oras-project/oras-go/blob/main/content/oci/oci.go#L195-L200).